packages/hash/frontend/src/pages/_app.page.tsx

```diff
@@ -57,35 +62,27 @@ const App: FunctionComponent<AppProps> = ({
     setSsr(false);
   }, []);
 
+  const { authenticatedUser } = useAuthInfo();
+
   useEffect(() => {
-    // If the user is logged in but hasn't completed signup and isn't on the signup page...
-    if (
-      authenticatedUser &&
-      !authenticatedUser.accountSignupComplete &&
-      !router.pathname.startsWith("/signup")
-    ) {
-      // ...then redirect them to the signup page.
-      void router.push("/signup");
-      // If the user is logged out redirect them to the login page
-    } else if (
-      !loading &&
-      !authenticatedUser &&
-      !(
-        router.pathname.startsWith("/login") ||
-        router.pathname.startsWith("/signup")
-      )
-    ) {
-      if (kratosSession) {
-        /**
-         * If we have a kratos session, but could not get the authenticated user,
-         * the kratos session may be invalid so needs to be re-fetched before redirecting.
-         */
-        void refetch().then(() => router.push("/login"));
-      } else {
-        void router.push("/login");
+    Sentry.configureScope((scope) => {
+      const sentryUser = scope.getUser();
+      if (!authenticatedUser && sentryUser) {
+        scope.setUser(null);
+      } else if (
+        authenticatedUser &&
+        sentryUser?.id !== authenticatedUser.entityEditionId.baseId
+      ) {
+        const primaryEmail = authenticatedUser.emails.find(
+          (email) => email.primary,
+        );
+        Sentry.setUser({
+          id: authenticatedUser.entityEditionId.baseId,
+          email: primaryEmail?.address,
+        });
```

Comment:
Can this be in a separate folder to keep `_app` easier to read?

Reply:
I've encapsulated this in a function and moved it into a separate file as part of https://github.com/hashintel/hash/pull/1637/commits/22b40db8e145c4776c7a95c1a1099ed6010bc4f6